External/SPMmodified/xASL/xASL_io_SaveNifti.m

```diff
@@ -60,20 +70,45 @@ function xASL_io_SaveNifti(pathOrigNifti, pathNewNifti, imNew, nBits, bGZip, cha
 	end
 end
 
+% JSON saving
+% Options are:
+% 1. bSaveJson     = false                       -> we don't save a new JSON sidecar
+% 2. bCopyOrigJson = true  & JsonFields = empty  -> we copy the original JSON sidecar only
+% 3. bCopyOrigJson = false & JsonFields = filled -> we create a new JSON sidecar with new fields only
+% 4. bCopyOrigJson = true  & JsonFields = filled -> we merge the original JSON sidecar with the new fields
+%                                                   the new fields have priority
+bSaveJson = false;
+
+if nargin < 7 || isempty(bCopyOrigJson)
+    bCopyOrigJson = false;
+else
+    bSaveJson = true;
+end
+
+if nargin < 8 || isempty(JsonFields)
+    JsonFields = struct([]);

```

Comment:
struct([]) -> struct().
Cause this commands as it is creates an array and you can't add new fields to it - see also below.

Reply:
Fixed in 67d6fdaccd87237de371db497c72181e2c8b1c8a